csi/controllerserver.py

```diff
@@ -131,6 +131,21 @@ def CreateVolume(self, request, context):
         pvtype = PV_TYPE_SUBVOL
         is_block = False
 
+        # Debug request.parameters
+        logging.info(logf(
+            "Request Parameters",
+            parameters=request.parameters
+        ))
+        storage_options = None
+        if request.parameters.get("storage_options"):
+            storage_options = request.parameters.get("storage_options")
```

Comment:
Storage options are given in the form of map in string, as map is not allowed. Is this OK?
```yaml
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: kadalu.storage-pool-1-vol-options
provisioner: kadalu
parameters:
  storage_name: "storage-pool-1"
  storage_options:
    '{
      "cluster/replicate":
      {
        "data-self-heal": "off", "entry-self-heal": "off"
      },
      "performance/open-behind":
      {
        "open-behind": "off"
      }
    }'
```

Reply:
- I couldn't say it's appealing or easy
- How about something like below, not mandatory to be ack'd though 
``` yaml
vol_ops:
  - cluster:
      - replicate:
          data-self-heal: off
          entry-self-heal: off
  - performance:
      - open-behind:
          open-behind: off
```
- Issue might be `off` will be interpreted as `true`

Reply:
We should use flat structure and Gluster users are familiar with that option names. For some options we may need to add support for aliases.

```yaml
storage_options:
  replicate.data-self-heal: off
  performance.open-behind: off
```

Split the Option name by `.` and then construct the dict when required.

Reply:
> as map is not allowed

Where is this not allowed? In Storage class or while adding to Volume Context?

Reply:
strorage class does not allow. 
```bash
error: error validating "vol-options-sc.yaml": error validating data: ValidationError(StorageClass.parameters.storage_options): invalid type for io.k8s.api.storage.v1.StorageClass.parameters: got "map", expected "string"; if you choose to ignore these errors, turn validation off with --validate=false
```